examples/map/map_segment.py

```diff
@@ -0,0 +1,121 @@
+"""
+=======================================================
+Segmenting a Map based on transformation of coordinates
+=======================================================
+
+A demonstration extracting a region of a particular map based on
+world coordinates in different systems.
+"""
+import matplotlib.pyplot as plt
+import numpy as np
+
+import astropy.units as u
+from astropy.coordinates import SkyCoord
+
+import sunpy.map
+from sunpy.coordinates import NorthOffsetFrame
+from sunpy.data.sample import AIA_171_IMAGE
+
+######################################################################
+# We start with the sample data.
+smap = sunpy.map.Map(AIA_171_IMAGE)
+
+######################################################################
+# A utility function gives us access to the helioprojective coordinate of each
+# pixel. From this we then transform the coordinates to the required frame.
+# For this example we are going to extract a region based on the
+# heliographic Stonyhurst coordinates, so we transform to that frame.
+all_hpc = sunpy.map.all_coordinates_from_map(smap)
+all_hgs = all_hpc.transform_to("heliographic_stonyhurst")
+
+######################################################################
+# Let's then segment the data based on coordinates and create a boolean mask
+# where True indicates invalid or deselected data.
+# This is the convention used by the `numpy.ma` module.
+segment = np.logical_or(
+    all_hgs.lon >= 50 * u.deg,
+    all_hgs.lon <= 0 * u.deg)
+
+######################################################################
+# Let's plot the segment separately, we create a new map with the segment as the
+# mask.
+# Numpy's masked arrays allow for a combination of standart numpy array and
+# a mask array. When an element of the mask is True, the corresponding element
+# of the associated array is said to be masked (invalid).
+# More information about Numpy's masked arrays :
+# https://numpy.org/doc/stable/reference/maskedarray.generic.html#using-numpy-ma
+# From the segment, we now mask out all values where `all_hgs.lon` is
+# NaN.
+segment = np.logical_or(
+    segment, np.isnan(
+        all_hgs.lon))
```

Comment:
```suggestion
segment = np.logical_or(segment, np.isnan(all_hgs.lon))
```

Reply:
This whole file is rather linebreak happy. Are you running something like black on it?

Reply:
I am not sure that black would do that normally.

Reply:
I've been using autopep8, that was because of the `np.logical_not(np.isnan())`.
I shall make this change.